src/plugins/intel_gpu/src/graph/graph_optimizer/add_required_reorders.cpp

```diff
@@ -326,4 +326,42 @@ void add_required_reorders::run(program& p) {
                         " (format: ", original_layout.format.to_string(),
                         ", data_type: ", ov::element::Type(original_layout.data_type), ") ");
     }
+
+    for (auto node : p.get_processing_order()) {
+        if (node->is_type<input_layout>()) {
+            if (node->get_output_layout().data_type == data_types::i16 ||
+                node->get_output_layout().data_type == data_types::u16 ||
+                node->get_output_layout().data_type == data_types::u32) {
+                auto supported_data_type = cldnn::element_type_to_data_type(node->get_output_layout().data_type);
+                layout reorder_layout = node->get_output_layout();
+                reorder_layout.data_type = supported_data_type;
+
+                auto new_reorder = std::make_shared<reorder>(node->id() + "_convert_type", node->id(), reorder_layout);
```

Comment:
Why is that needed here? I'd expect that `ov::pass::ConvertPrecision` inserts explicit `Convert` operations for inputs since we request conversion from most of integer types to i32.

Reply:
The purpose of this PR is not only to support element types i16, u16, and u32, but also to share user tensors allocated as cl_mem directly as plugin tensors. Of course I have tried as you suggested, but when creating primitive for Parameter op and Result op within the GPU Plugin, it converts to the element type supported by the plugin, so adding Convert is meaningless. So, I modified cldnn::input_layout to support those data types and chose to add a reorder at the next. In the case of Result op, it is already a reorder, so no additional processing is needed.

Reply:
Well, I understand the part which allows these types for input_layout primitive. But my point is that we run ConverPrecision pass already for u16/i16/u32 -> i32 conversion which means that for Parameter/Result ops with these types we'll have explicit Convert operation on the ngraph level which will produce reorder in cldnn graph. So I don't understand why you try to insert yet another reorder in this `add_required_reorders` pass

Reply:
And the test that you've added doesn't really check that this pass works correctly as original graph contains u16 -> f32 convert right after Parameter op.
![image](https://github.com/user-attachments/assets/f9631b8b-55b7-4d14-b1cb-371e44e4e019)

So I think we need a test with u16 Parameter followed by any op which can process u16 tensor directly (Some elementwise probably), i.e. original graph shouldn't contain any type conversions for a part of the ops.

Reply:
How about this graph for test?
![image](https://github.com/user-attachments/assets/a0e49013-28ad-406d-9ba9-6a90d9bb306c)
![image](https://github.com/user-attachments/assets/b457c7ce-7185-433a-b990-1fa6b609e7fe)